ReportPortal.ATM/TestProject/Pages/AddNewWidgetPage/AddNewWidgetPage.locator.cs

```diff
@@ -0,0 +1,21 @@
+﻿using OpenQA.Selenium;
+
+namespace TestProject.Pages.AddNewWidgetPage;
+
+public partial class AddNewWidgetPage
+{
+    private By _nextStepButtonLocator = By.XPath("//span[text()='Next step']");
+    private By _defaultFilterSelector = By.ClassName("inputRadio__toggler--ygpdQ");
+    private By _widgetNameInputLocator = By.XPath("//input[@placeholder='Enter widget name']");
+    private By _addWidgetButtonLocator = By.XPath("//button[text()='Add']");
+    private By _widgetTypeLocator = By.CssSelector("widgetHeader__type--yZiVg");
+    private By _widgetNamesLocator = By.CssSelector(".widgetHeader__widget-name-block--AOAHS");
+
+    public IWebElement WidgetTypeInput(string type) => _webDriver.
+        FindElement(By.XPath($"//div[text()='{type}']/parent::span/preceding-sibling::span"));
```

Comment:
Never parameterize locators, and web elements should always be properties, not methods.

You can have an `IList<IWebElements>` to represent all widgets, then, you can have a partial locator to target the widget's parent's preceding sibling's span.

.locators.cs
```
private By _widgetListLocator = By.XPath("//div[@class = 'something-that-locates-all-widgets']");
private By _precedingSiblingContainerLabelLocator = By.XPath(".//parent::span/preceding-sibling::span");

private IList<IWebElement> WidgetList => _webDriver.FindElement(_widgetListLocator);
```

.private.cs
```
private void ClickWidgetLabel(string label)
{
    try
    {
        ArgumentException.ThrowIfNullOrWhiteSpace(label);

        _webDriver.WaitUntilElementIsVisible(_widgetTypeLocator,
                    _webDriverService.GetWebDriverConfiguration().LongTimeout,
                    _webDriverService.GetWebDriverConfiguration().PollingIntervalTimeout);

        WidgetList
            .First(widget => widget.Text.Equals(label, StringComparison.OrdinalIgnoreCase))
            .FindElement(_precedingSiblingContainerLabelLocator)
            .Click();
    }
    catch (Exception e)
    {
        throw;
    }
}
```